ivy/src/main/scala/sbt/IvyInterface.scala

```diff
@@ -21,9 +22,9 @@ final case class ModuleInfo(nameFormal: String, description: String = "", homepa
 }
 
 /** Basic SCM information for a project module */
-final case class ScmInfo(browseUrl: URL, connection: String, devConnection: Option[String] = None)
+final case class ScmInfo(browseUrl: URL, connection: String, devConnection: Option[String] = None, tag: Option[String] = None)
```

Comment:
Unfortunately this isn't a binary compatible change.   

You'll need to rip the apply/unapply methods into a companion object and change it to a regular class.  Then have two constructors, one with the existing options and one with the extra.  It's really gunky.

An alternative MAY be to have the tag encoded in the URL as a "#" or something.  Not the best., but it would work.